src/internal/adapter/web/WebPresentationModeManager.ts

```diff
@@ -24,57 +24,47 @@ export class WebPresentationModeManager {
     }
 
-    this.prepareForPresentationModeChanges();
+    this.maybePrepareForPresentationModeChanges();
 
     if (fullscreenAPI !== undefined) {
-      // All other browsers
+      // If the browser supports the fullscreenAPI, put the element that encloses the player & UI in fullscreen.
       if (presentationMode === PresentationMode.fullscreen) {
-        const appElement = document.getElementById('app') ?? document.getElementById('root');
+        const appElement = document.getElementById('theoplayer-root-container');
         if (appElement !== null) {
-          const promise = appElement[fullscreenAPI.requestFullscreen_]();
-          if (promise && promise.then) {
-            promise.then(noOp, noOp);
-          }
+          appElement[fullscreenAPI.requestFullscreen_]()?.then?.(noOp, noOp);
         }
       } else if (presentationMode === PresentationMode.pip) {
         this._player.presentation.requestMode('native-picture-in-picture');
       } else {
         if (this._presentationMode === PresentationMode.fullscreen) {
-          const promise = document[fullscreenAPI.exitFullscreen_]();
-          if (promise && promise.then) {
-            promise.then(noOp, noOp);
-          }
+          document[fullscreenAPI.exitFullscreen_]()?.then?.(noOp, noOp);
         }
         if (this._presentationMode === PresentationMode.pip) {
-          void document.exitPictureInPicture();
+          document.exitPictureInPicture().then(noOp, noOp);
         }
       }
     } else {
-      // iOS Safari doesn't properly support fullscreen, use native fullscreen instead
+      // Some browsers, like iOS Safari, can only put a videoElement in fullscreen; let the player decide which one.
       if (presentationMode === PresentationMode.fullscreen) {
-        this._element?.webkitEnterFullscreen?.();
+        this._player.presentation.requestMode(PresentationMode.fullscreen);
       } else if (presentationMode === PresentationMode.pip) {
         this._player.presentation.requestMode('native-picture-in-picture');
       } else {
-        this._element?.webkitSetPresentationMode?.(PresentationMode.inline);
+        this._player.presentation.requestMode(PresentationMode.inline);
       }
     }
   }
 
-  private prepareForPresentationModeChanges() {
-    const elements = this._player.element.children;
-    for (const element of Array.from(elements)) {
-      if (element.tagName === 'VIDEO') {
-        const videoElement = element as HTMLVideoElement;
-        if ((videoElement.src !== null && videoElement.src !== '') || videoElement.srcObject !== null) {
-          this._element = videoElement;
-          break;
-        }
-      }
+  private maybePrepareForPresentationModeChanges() {
+    if (this._didPrepare) {
+      return;
     }
+    this._didPrepare = true;
+
     // listen for fullscreen updates on document
     if (fullscreenAPI !== undefined) {
       document.addEventListener(fullscreenAPI.fullscreenchange_, this.updatePresentationMode);
```

Comment:
Don't we also need eventlisteners for the native pip here?

Reply:
The PiP events come from the player as we now always ask the player to go into PiP